File: src/main/java/com/choru/lock/domain/Order.java
package com.choru.lock.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Table(name = "orders")
@NoArgsConstructor
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;
    private Long productId;
    private Long couponId;

    public Order(Long userId, Long productId, Long couponId) {
        this.userId = userId;
        this.productId = productId;
        this.couponId = couponId;
    }
}

File: src/main/kotlin/com/choru/lock/domain/Order.kt
package com.choru.lock.domain

import jakarta.persistence.*

@Entity
@Table(name = "orders")
class Order(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    val userId: Long,

    val productId: Long,

    val couponId: Long
) {
    constructor(userId: Long, productId: Long, couponId: Long) : this(
        id = null,
        userId = userId,
        productId = productId,
        couponId = couponId
    )
}
